### DISCUSSION TITLE
Disappear characters in replacement word

### DISCUSSION THREAD

**Discussion by Muhammadraafat1** [pymupdf/PyMuPDF#3667] [Looking for help] [answered]
Hello, 
Here the original text:
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/ca21c6e5-fb64-4782-99df-c1ec5738349b)

When I made annotation it looks good:
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/0b3e4dc1-ea11-40ba-aa81-e223ee80f6bc)

When I applied redaction it's not good:
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/5b747f7c-f6fc-49ef-a9bd-cdf85c835002)

I tried to replace the Graham Road only and it works well as appear here:
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/d1856703-8b54-400a-8e76-223a3f4c493c)

I can't understand the problem is in OCRed PDF? or in insert_text! I had this issue in another PDF but it solved by fitz.TOOLS.set_small_glyph_heights(True), in this PDF nothing change, can you recommend something to solve?

**Comment by JorjMcKie** [comment, marked as answer]
The problem is that the redaction algorithm does not (and **_must not!_**) care about small glyph heights setting. It must in contrast make sure (for legal reasons) that not a single bit remains that might carry information to be removed.

To solve your issue, you can always further decrease the height of your redaction rectangle. For example make a small horizontal rectangular stripe around the middle line. As long as all desired characters overlap that stripe in any way, they will be removed.
